### PR TITLE
KAFKA-5686: update compression docs

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
minor change in compression docs. removed reference to the old link.
cc @ijuma

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6842/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6828/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by omkreddy** [comment]
cc @ijuma  ping for review

File: docs/design.html [hunk 1]
```diff
@@ -136,7 +136,7 @@
     Kafka supports this with an efficient batching format. A batch of messages can be clumped together compressed and sent to the server in this form. This batch of messages will be written in compressed form and will
     remain compressed in the log and will only be decompressed by the consumer.
     <p>
-    Kafka supports GZIP, Snappy and LZ4 compression protocols. More details on compression can be found <a href="https://cwiki.apache.org/confluence/display/KAFKA/Compression">here</a>.
+    Kafka supports GZIP, Snappy and LZ4 compression protocols. Compression related config properties can be found in the <a href="#producerconfigs">Producer configs</a> section.
```
Comment (ijuma) [inline docs/design.html]: Hmm, not sure about this. Compression related config properties can be found in the producer and broker configs, right?

Comment (omkreddy) [reply docs/design.html]: @ijuma  yes.  mentioned broker configs in the compression docs.

Comment (omkreddy) [reply docs/design.html]: @ijuma pinging for review

**Comment by aisven** [comment]
I just tried to set `LZ4` in the configuration of Kafka Broker which leads to a fatal that says it should be lower case `lz4`.

```
[2017-12-29 13:55:15,393] FATAL [KafkaServer id=1] Fatal error during KafkaServer startup. Prepare to shutdown (kafka.server.KafkaServer)
org.apache.kafka.common.config.ConfigException: Invalid value LZ4 for configuration compression.type: String must be one of: uncompressed, snappy, lz4, gzip, producer
at org.apache.kafka.common.config.ConfigDef$ValidString.ensureValid(ConfigDef.java:897)
at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:469)
at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:453)
at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:62)
at kafka.log.LogConfig.<init>(LogConfig.scala:68)
at kafka.log.LogManager$.apply(LogManager.scala:783)
at kafka.server.KafkaServer.startup(KafkaServer.scala:222)
at io.confluent.support.metrics.SupportedServerStartable.startup(SupportedServerStartable.java:112)
at io.confluent.support.metrics.SupportedKafka.main(SupportedKafka.java:58)
```

Please consider this difference between API and Broker-Configuration.

**Pull request closed without merging**